src/stores/deck.js

```diff
@@ -23,23 +20,218 @@ function shuffleDeck(deck) {
+    set({ deck: shuffleDeck([...baseDeck, ...extras]) });
+  },
+
+  buildAndShuffle: deckId => {
+    const state = get();
+    const id = deckId || state.activeDeckId;
+    const selected = state.decks[id] || { id, ...defaultsFor(id), loadout: [] };
+    const extras = selected.loadout.map(mapItemToCard);
+    set({
+      activeDeckId: id,
+      decks: { ...state.decks, [id]: selected },
+      loadout: selected.loadout,
+      deck: shuffleDeck([...state.baseDeck, ...extras]),
+    });
+  },
+
+  addOwnedShopCard: item =>
+    set(state => {
+      if (state.ownedShopCards.find(x => x.id === item.id)) return state;
+      return { ownedShopCards: [...state.ownedShopCards, item] };
+    }),
+
+  addToLoadout: id =>
+    set(state => {
+      const deckId = state.activeDeckId;
+      const selected = state.decks[deckId] || {
+        id: deckId,
+        ...defaultsFor(deckId),
+        loadout: [],
+      };
+      if (selected.loadout.length >= 10) return state;
+      if (selected.loadout.find(x => x.id === id)) return state;
+      const item = state.ownedShopCards.find(x => x.id === id);
+      if (!item) return state;
+      const nextLoadout = [...selected.loadout, item];
+      return {
+        decks: { ...state.decks, [deckId]: { ...selected, loadout: nextLoadout } },
+        loadout: nextLoadout,
+      };
+    }),
+
+  removeFromLoadout: id =>
+    set(state => {
+      const deckId = state.activeDeckId;
+      const selected = state.decks[deckId] || {
+        id: deckId,
+        ...defaultsFor(deckId),
+        loadout: [],
+      };
+      const nextLoadout = selected.loadout.filter(x => x.id !== id);
+      return {
+        decks: { ...state.decks, [deckId]: { ...selected, loadout: nextLoadout } },
+        loadout: nextLoadout,
+      };
+    }),
+
+  addToLoadoutFor: (deckId, id) =>
+    set(state => {
+      const key = String(deckId);
+      const base = state.decks[key] || { id: key, ...defaultsFor(key), loadout: [] };
+      if (base.loadout.length >= 10) return state;
+      if (base.loadout.find(x => x.id === id)) return state;
+      const item = state.ownedShopCards.find(x => x.id === id);
+      if (!item) return state;
+      const nextLoadout = [...base.loadout, item];
+      const updatedDecks = { ...state.decks, [key]: { ...base, loadout: nextLoadout } };
+      const syncActive = key === state.activeDeckId ? { loadout: nextLoadout } : {};
+      return { decks: updatedDecks, ...syncActive };
+    }),
+
+  removeFromLoadoutFor: (deckId, id) =>
+    set(state => {
+      const key = String(deckId);
+      const base = state.decks[key] || { id: key, ...defaultsFor(key), loadout: [] };
+      const nextLoadout = base.loadout.filter(x => x.id !== id);
+      const updatedDecks = { ...state.decks, [key]: { ...base, loadout: nextLoadout } };
+      const syncActive = key === state.activeDeckId ? { loadout: nextLoadout } : {};
+      return { decks: updatedDecks, ...syncActive };
+    }),
```

Comment:
Yeah after looking at this file further I think we should remove all the code that was built to support multiple decks. We only have one deck right now. This work should be limited to adding new cards to the existing standard deck.